### PR TITLE
[Merged by Bors] - feat(Topology) define jacobson spaces

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/JacobsonSpace.lean [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+lemma closedPoints_eq_univ [T2Space X] :
+    closedPoints X = Set.univ :=
+  Set.eq_univ_iff_forall.mpr fun _ ↦ isClosed_singleton
+
+end closedPoints
+
+/-- The class of jacobson spaces, i.e.
+spaces such that the set of closed points are dense in every closed subspace. -/
+@[mk_iff]
```
Comment (jcommelin) [inline Mathlib/Topology/JacobsonSpace.lean:59]:
```suggestion
@[mk_iff, stacks 005U]
```

File: Mathlib/Topology/JacobsonSpace.lean [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  have inst : Finite X := Set.finite_univ_iff.mp (this ▸ h)
+  rw [← forall_open_iff_discrete]
+  intro s
+  rw [← isClosed_compl_iff, ← sᶜ.biUnion_of_singleton]
+  refine sᶜ.toFinite.isClosed_biUnion fun x _ ↦ ?_
+  rw [← mem_closedPoints_iff, this]
+  trivial
+
+instance (priority := 100) [Finite X] [JacobsonSpace X] : DiscreteTopology X :=
+  JacobsonSpace.discreteTopology (Set.toFinite _)
```
Comment (jcommelin) [inline Mathlib/Topology/JacobsonSpace.lean:144]: Could you please also add the instance in the other direction? Discrete => Jacobson

Comment (erdOne) [reply Mathlib/Topology/JacobsonSpace.lean:144]: Okay I added T2 => Jacobson

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by erdOne** [comment]
bors merge

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**